azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -1255,4 +1266,165 @@ public void accept(MessageSender a) {
+							(String)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_OFFSET),
+							((Date)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_TIME_UTC)).toInstant()));
+					return future2;
+				}
+	        });
+        }
+        
+        return future1;
+    }
+    
+    private <T> CompletableFuture<T> addManagementToken(Map<String, String> request)
+    {
+    	CompletableFuture<T> retval = null;
+        try {
+        	String audience = String.format("amqp://%s/%s", this.underlyingFactory.getHostName(), this.eventHubName);
+        	String token = this.underlyingFactory.getTokenProvider().getToken(audience, ClientConstants.TOKEN_REFRESH_INTERVAL); 
+			request.put(ClientConstants.MANAGEMENT_SECURITY_TOKEN_KEY, token);
+		} 
+        catch (InvalidKeyException | NoSuchAlgorithmException | IOException e) {
+        	retval = new CompletableFuture<T>();
+        	retval.completeExceptionally(e);
+		}
+    	return retval;
+    }
+    
+    private CompletableFuture<Map<String, Object>> managementWithRetry(Map<String, String> request) {
+        Instant endTime = Instant.now().plus(this.underlyingFactory.getOperationTimeout());
+        CompletableFuture<Map<String, Object>> rawdataFuture = new CompletableFuture<Map<String, Object>>();
+        
+        ManagementRetry retrier = new ManagementRetry(rawdataFuture, endTime, this.underlyingFactory, request, 0, null);
+        Timer.schedule(retrier, Duration.ZERO, TimerType.OneTimeRun);
+        
+        return rawdataFuture;
+    }
+    
+    private class ManagementRetry implements Runnable {
+    	private final CompletableFuture<Map<String, Object>> finalFuture;
+    	private final Instant endTime;
+    	private final MessagingFactory mf;
+    	private final Map<String, String> request;
+    	private final int retryCount;
+    	private final String clientId;
+    	
+    	public ManagementRetry(CompletableFuture<Map<String, Object>> future, Instant endTime, MessagingFactory mf,
+    			Map<String, String> request, int retryCount, String clientId) {
+    		this.finalFuture = future;
+    		this.endTime = endTime;
+    		this.mf = mf;
+    		this.request = request;
+    		this.retryCount = retryCount;
+    		if (clientId != null) {
+    			this.clientId = clientId;
+    		}
+    		else {
+    			this.clientId = UUID.randomUUID().toString();
+    		}
+    	}
+    	
+		@Override
+		public void run() {
+			CompletableFuture<Map<String, Object>> intermediateFuture = this.mf.getManagementChannel().request(this.mf.getReactorScheduler(), request);
+			intermediateFuture.whenComplete(new BiConsumer<Map<String, Object>, Throwable>() {
+				@Override
+				public void accept(Map<String, Object> result, Throwable error) {
+					if ((result != null) && (error == null)) {
+						ManagementRetry.this.finalFuture.complete(result);
+					}
+					else {
+						int waitSeconds = (ManagementRetry.this.retryCount + 1) * (ManagementRetry.this.retryCount + 1);
```

Comment:
>waitSeconds [](start = 10, length = 11)

use mf.getRetryPolicy().getNextRetryInterval(exception) - which will apply the retryPolicy - with your current logic - even if user have retry policy configured as NoRetry - we will go ahead and retry..